java/client/src/main/java/com/youtube/vitess/client/cursor/SimpleCursor.java

```diff
@@ -59,10 +59,7 @@ public boolean next() throws SQLException {
   }
 
   @Override
-  protected Row getCurrentRow() throws SQLException {
-    if (row == null) {
-      throw new SQLDataException("no current row");
-    }
+  public Row getCurrentRow() {
```

Comment:
Then what about get rid of getCurrentRow method and let "next" method return a row object if succeeded, return null if reach to the end of the stream and throw exception if error happens.